lib/salus/scanners/report_go_dep.rb

```diff
@@ -7,38 +7,126 @@
 module Salus::Scanners
   class ReportGoDep < Base
     def run
-      unless should_run?
+      if @repository.go_sum_present?
+        record_dep_from_go_sum
+      elsif @repository.go_mod_present?
+        record_dep_from_go_mod
+      elsif @repository.dep_lock_present?
+        record_dep_from_go_lock_package
+      else
         raise(
           InvalidScannerInvocationError,
-          'Cannot report on Go dependencies without a Gopkg.lock file.'
+          'Cannot report on Go dependencies without a Gopkg.lock, go.mod, or go.sum file'
         )
       end
+    end
+
+    def record_dep_from_go_sum
+      go_sum_path = "#{@repository.path_to_repo}/go.sum"
+      dep_list = []
+
+      File.open(go_sum_path).each("=\n") do |line|
+        dep_list.append(
+          {
+            "fullDependency" => line,
+          "name" => get_name(line),
```

Comment:
if you made one function to parse the line info you could avoid splitting the string twice.

Reply:
Refactored for less splitting 👍